### PR TITLE
Update: [AEA-3806] - add linting to GitHub actions

### PR DESCRIPTION
**Pull request by seansteberisal**
## Summary

**Remove items from this list if they are not relevant. Remove this line once this has been done**

- Routine Change

### Details

actionlint added to pre-commit to ensure linting is enforced on github actions .yml files

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,8 +26,6 @@ jobs:
   tag_release:
     needs: quality_checks
     runs-on: ubuntu-latest
-    outputs:
```
Comment (originalphil) [inline .github/workflows/ci.yml:29]: Does this want output_version_tag like in release.yml?

File: .github/workflows/pr-link.yml [hunk 2]
```diff
@@ -5,20 +5,22 @@ on:
...
+        run: echo "$REF" | grep -i -E -q "(aea-[0-9]+)|(dependabot\/)"
         continue-on-error: true
 
       - name: Grab ticket name
-        if: contains(github.event.pull_request.head.ref, 'aea-') || contains(github.event.pull_request.head.ref, 'AEA-') || contains(github.event.pull_request.head.ref, 'apm-') || contains(github.event.pull_request.head.ref, 'APM-') || contains(github.event.pull_request.head.ref, 'apmspii-') || contains(github.event.pull_request.head.ref, 'APMSPII-') || contains(github.event.pull_request.head.ref, 'adz-') || contains(github.event.pull_request.head.ref, 'ADZ-') || contains(github.event.pull_request.head.ref, 'amb-') || contains(github.event.pull_request.head.ref, 'AMB-')
-        run: echo ::set-env name=TICKET_NAME::$(echo ${{ github.event.pull_request.head.ref }} | grep -i -o '\(aea-[0-9]\+\)\|\(apm-[0-9]\+\)\|\(apmspii-[0-9]\+\)\|\(adz-[0-9]\+\)|\(amb-[0-9]\+\)' | tr '[:lower:]' '[:upper:]')
+        if: contains(github.event.pull_request.head.ref, 'aea-') || contains(github.event.pull_request.head.ref, 'AEA-')
```
Comment (originalphil) [inline .github/workflows/pr-link.yml:16]: Should $REF also be used in here?

File: .github/workflows/pr-link.yml [hunk 3]
```diff
@@ -5,20 +5,22 @@ on:
...
+        run: echo name=TICKET_NAME::"$(echo "$REF" | grep -i -o '\(aea-[0-9]\+\)' | tr '[:lower:]' '[:upper:]')" >> "$GITHUB_ENV"
         continue-on-error: true
         env:
           ACTIONS_ALLOW_UNSECURE_COMMANDS: true
 
       - name: Comment on PR with link to JIRA ticket
-        if: contains(github.event.pull_request.head.ref, 'aea-') || contains(github.event.pull_request.head.ref, 'AEA-') || contains(github.event.pull_request.head.ref, 'apm-') || contains(github.event.pull_request.head.ref, 'APM-') || contains(github.event.pull_request.head.ref, 'apmspii-') || contains(github.event.pull_request.head.ref, 'APMSPII-') || contains(github.event.pull_request.head.ref, 'adz-') || contains(github.event.pull_request.head.ref, 'ADZ-') || contains(github.event.pull_request.head.ref, 'amb-') || contains(github.event.pull_request.head.ref, 'AMB-')
+        if: contains(github.event.pull_request.head.ref, 'aea-') || contains(github.event.pull_request.head.ref, 'AEA-')
```
Comment (originalphil) [inline .github/workflows/pr-link.yml:23]: Should $REF also also be used in here?

File: .github/workflows/regression_tests.yml [hunk 4]
```diff
@@ -76,10 +76,10 @@ jobs:
         uses: actions/cache@a2ed59d39b352305bdd2f628719a53b2cc4f9613
         id: cache-venv
         with:
-          path: ./.venv/
-          key: ${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-venv-${{ hashFiles('pyproject.toml') }}
+          path: .venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('pyproject.toml') }}
```
Comment (originalphil) [inline .github/workflows/regression_tests.yml:80]: Have these changes been tested?

Comment (seansteberisal) [reply .github/workflows/regression_tests.yml:80]: This doesn't work either way. I have a tech-debt ticket to address this:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (seansteberisal) [reply .github/workflows/ci.yml:29]: added

File: .github/workflows/pr-link.yml (hunk 2 above)
Comment (seansteberisal) [reply .github/workflows/pr-link.yml:16]: There doesn't seem to be a way to do this inside of the contains() method. I checked the EPS API and it's done the same way as this currently

**Pull request merged**